firedrake/preconditioners/asm.py

```diff
@@ -224,7 +237,7 @@ def get_patches(self, V):
                         continue
                     off = section.getOffset(p)
                     # Local indices within W
-                    W_indices = numpy.arange(off*W.value_size, W.value_size * (off + dof), dtype=IntType)
+                    W_indices = slice(off*W.value_size, W.value_size * (off + dof))
```

Comment:
This is just so you don't make a big array for indexing.

Reply:
Yes